etl/steps/data/garden/owid/2022-12-19/gdp_last_2000_years.py

```diff
@@ -0,0 +1,139 @@
+    df_wdi = df_wdi.sort_values(by=["year"]).reset_index(drop=True)
+
+    # Select GDP and GDP pc in international-$ in 2017 prices
+    df_wdi = df_wdi[["country", "year", "ny_gdp_mktp_pp_kd", "ny_gdp_pcap_pp_kd"]]
+    df_wdi = df_wdi.rename(columns={"ny_gdp_mktp_pp_kd": "gdp", "ny_gdp_pcap_pp_kd": "gdp_per_capita"})
+    # Filter "World" entity
+    df_wdi = df_wdi[df_wdi["country"] == "World"]
+    # Drop empty World GDP estimations
+    df_wdi = df_wdi.dropna().reset_index(drop=True)
+
+    # Load Maddison Dataset 2010 - garden step. It contains global GDP estimations between 1 and 1820
+    ds = Dataset(DATA_DIR / f"garden/ggdc/{VERSION_MADDISON2010}/maddison_database_2010")
+    df_maddison2010 = ds["maddison_database_2010"]
+    df_maddison2010 = pd.DataFrame(df_maddison2010)
+
+    # Select variables to evaluate and reference year to compare Maddison world estimations to World Bank WDI (1990)
+    var_list = ["gdp", "gdp_per_capita"]
+    year_ref = 1990
+
+    # Select only "World" entity in Maddison dataframe
+    df_world = df_maddison[df_maddison["country"] == "World"].copy().reset_index(drop=True)
+    df_world = df_world[df_world["year"] <= year_ref].reset_index(drop=True)
+
+    # Drop population, as it's not needed
+    df_world = df_world.drop(columns=["population"])
+
+    # Merge both Maddison world estimation and WDI world estimation
+    df = pd.merge(df_world, df_wdi, on="year", how="left", suffixes=("_maddworld", "_wdi"))
+
+    for var in var_list:
+        # The reference value for the world aggregation is the 1990 world aggregation value
+        value_ref_countries = df.loc[df["year"] == year_ref, f"{var}_maddworld"].iloc[0]
+        # The scalar is this reference value divided by each each world aggregation
+        df[f"{var}_scalar"] = value_ref_countries / df[f"{var}_maddworld"]
+
+        # The reference value is the 1990 WDI value
+        value_ref = df.loc[df["year"] == year_ref, f"{var}_wdi"].iloc[0]
+        # The rest of the estimated values are the division between the reference value and the scalars
+        df[f"{var}_estimated"] = value_ref / df[f"{var}_scalar"]
+
+        # Rename the estimated to gdp/gdp_per_capita
+        # Round depending on the variable
+        if var == "gdp":
+            df[f"{var}"] = np.where(df[f"{var}_wdi"].isnull(), df[f"{var}_estimated"].round(0), df[f"{var}_wdi"])
+        else:
+            df[f"{var}"] = np.where(df[f"{var}_wdi"].isnull(), df[f"{var}_estimated"], df[f"{var}_wdi"])
+
+    # Specify "World" entity for each row
+    df["country"] = "World"
+
+    # Select only country, year and the new gdp and gdp_per_capita columns, filtering the year 1990
+    df = df[["country", "year"] + var_list + ["gdp_maddworld", "gdp_per_capita_maddworld"]]
+
+    # Concatenate WDI results with estimation
+    df = pd.concat([df, df_wdi[df_wdi["year"] > year_ref]], ignore_index=True)
+
+    # Specify new reference year for the 1 - 1820 period
+    year_ref = 1820
```

Comment:
It would be better to pack this long processing into smaller functions that do a specific thing.